### PR TITLE
[plugin.video.ted.talks] 4.3.1

### PR DESCRIPTION
**Pull request by moreginger**
### Description

Various fixes (plugin currently broken) as per changelog.

Dropping support for <Krypton. Clean up MR to follow.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.ted.talks/.pylintrc [hunk 1]
```diff
@@ -0,0 +1,407 @@
+[MASTER]
```
Comment (Rechi) [inline plugin.video.ted.talks/.pylintrc]: drop this file from the PR

File: plugin.video.ted.talks/addon.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <import addon="script.module.parsedom" version="2.5.2"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="default.py">
+    <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <!-- There are subtitles on TED in many languages,
+    and this plugin is (partially) internationalized in several,
+    but it seems better to claim the one we are sure of for now. -->
+    <language>en_GB</language>
```
Comment (Rechi) [inline plugin.video.ted.talks/addon.xml]: `en` instead of `en_GB`

File: plugin.video.ted.talks/changelog.txt [hunk 3]
```diff
@@ -0,0 +1,141 @@
+Version 4.3.1
```
Comment (Rechi) [inline plugin.video.ted.talks/changelog.txt]: you may want to add a news section in addon.xml as changelog.txt isn't read in Kodi Krypton and newer.

File: plugin.video.ted.talks/resources/lib/menu_util_test.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import unittest
```
Comment (Rechi) [inline plugin.video.ted.talks/resources/lib/menu_util_test.py]: It's great that you have tests, but those are not required for running the add-on in Kodi. So please drop test related files from the PR.

File: plugin.video.ted.talks/resources/lib/ted_talks.py [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+    def addItem(self, title, mode, url=None, img='', args={}, video_info={}, isFolder=True, total_items=0):
+        # Create action url
+        args['mode'] = mode
+        if url:
+            args['url'] = url
+
+        args = [k + '=' + urllib.quote_plus(v.encode('ascii', 'ignore')) for k, v in args.iteritems()]
+        action_url = sys.argv[0] + '?' + "&".join(args)
+
+        li = xbmcgui.ListItem(label=title, iconImage=img, thumbnailImage=img)
```
Comment (Rechi) [inline plugin.video.ted.talks/resources/lib/ted_talks.py]: `iconImage` and `thumbnailImage` are deprecated, use `setArt(...)`

File: plugin.video.ted.talks/resources/lib/ted_talks.py [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+            li.addContextMenuItems([], replaceItems=False)
+        xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=action_url, listitem=li, isFolder=isFolder, totalItems=total_items)
+
+    def playVideo(self, url, icon):
+
+        plugin.report('%s = %s' % ('url', str(url)), level='debug')
+
+        subs_language = settings.get_subtitle_languages()
+        title, url, subs, info_labels = self.ted_talks.getVideoDetails(url=url, video_quality=settings.video_quality, subs_language=subs_language)
+        li = xbmcgui.ListItem(title, iconImage=icon, thumbnailImage=icon, path=url)
```
Comment (Rechi) [inline plugin.video.ted.talks/resources/lib/ted_talks.py]: `iconImage` and `thumbnailImage` are deprecated, use `setArt(...)`

File: plugin.video.ted.talks/resources/lib/ted_talks.py (hunk 5 above)
Comment (moreginger) [reply plugin.video.ted.talks/resources/lib/ted_talks.py]: Thanks for your feedback @Rechi. I've made the suggested improvements.

File: plugin.video.ted.talks/addon.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    <platform>all</platform>
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
+    <news>
+v4.3.1
+- Dropped support for Kodi versions older than 17 (Krypton)
+- High resolution plugin icon (pull#71 thanks to MarhyCZ)
+- Updates to support new TED site (pull#64 thanks to and3md)
+    </news>
+    <forum>https://forum.kodi.tv/showthread.php?tid=36866</forum>
+    <website>https://github.com/moreginger/xbmc-plugin.video.ted.talks</website>
```
Comment (Rechi) [inline plugin.video.ted.talks/addon.xml]: website should be the site which provides the content, something like `www.ted.com`

**Comment by Rechi** [comment]
@moreginger this add-on isn't Python 3 compatible. Future updates have to be compatible with Python 3.
Have a look at https://kodi.tv/article/attention-addon-developers-migration-python-3 for more information.

**Comment by moreginger** [comment]
Thanks @Rechi, noted that I need to get Python 3 sorted.

**Pull request merged**